gpii/node_modules/gpii-oauth2/gpii-oauth2-datastore/src/InMemoryDataStore.js

```diff
@@ -341,4 +372,51 @@ var fluid = fluid || require("infusion");
         return undefined;
     };
 
+    gpii.oauth2.dataStore.findCredentialClientTokenById = function (credentialClientTokens, id) {
+        return fluid.find_if(credentialClientTokens, function (cct) {
+            return cct.id === id;
+        });
+    };
+
+    gpii.oauth2.dataStore.findCredentialClientTokenByClientId = function (credentialClientTokens, clientId) {
+        var credentialClientToken = fluid.find_if(credentialClientTokens, function (cct) {
+            return cct.clientId === clientId && cct.revoked === false;
+        });
+        return credentialClientToken ? credentialClientToken : undefined;
+    };
+
+    gpii.oauth2.dataStore.findCredentialClientTokenByAccessToken = function (credentialClientTokens, accessToken) {
+        var credentialClientToken = fluid.find_if(credentialClientTokens, function (cct) {
+            return cct.accessToken === accessToken && cct.revoked === false;
+        });
+        return credentialClientToken ? credentialClientToken : undefined;
+    };
+
+    gpii.oauth2.dataStore.addCredentialClientToken = function (model, applier, credentialClientTokenData) {
+        var credentialClientTokenId = model.credentialClientTokensIdSeq;
+        applier.change("credentialClientTokensIdSeq", model.credentialClientTokensIdSeq + 1);
+        var credentialClientToken = {
+            id: credentialClientTokenId, // primary key
+            clientId: credentialClientTokenData.clientId, // foreign key
+            accessToken: credentialClientTokenData.accessToken,
+            allowAddPrefs: credentialClientTokenData.allowAddPrefs,
+            revoked: false
+        };
+        model.credentialClientTokens.push(credentialClientToken);
+        applier.change("credentialClientTokens", model.credentialClientTokens);
+        return credentialClientToken;
+    };
+
+    gpii.oauth2.dataStore.revokeCredentialClientToken = function (credentialClientTokens, applier, credentialClientTokenId) {
+        var credentialClientToken = gpii.oauth2.dataStore.findCredentialClientTokenById(credentialClientTokens, credentialClientTokenId);
+        if (credentialClientToken) {
+            credentialClientToken.revoked = true;
+            applier.change("credentialClientTokens", credentialClientTokens);
+        }
+    };
+
+    gpii.oauth2.dataStore.findCredentialClientByAccessToken = function (credentialClientTokens, clients, accessToken) {
```

Comment:
Regarding the renaming we discussed, I'm thinking that we call this one findClientByClientCredentialsAccessToken.